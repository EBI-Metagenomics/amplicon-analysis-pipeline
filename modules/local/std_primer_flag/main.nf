
process STD_PRIMER_FLAG {
    // Check for presence of standard library of primers (stored in ./data/standard_primers)
    tag "$meta.id"
    label 'light'
    conda "${projectDir}/conf/environment.yml"

    input:
    tuple val(meta), path(reads_merged)
    path(std_primer_library)

    output:
    tuple val(meta), path("*std_primers.fasta"), emit: std_primer_out
    path "*std_primer_out.txt"

    """
    standard_primer_matching -i $reads_merged -p $std_primer_library -s ${meta.id}_${meta.var_region} -o ./
    """

}
