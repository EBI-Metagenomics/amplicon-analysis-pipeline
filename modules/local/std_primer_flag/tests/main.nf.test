nextflow_process {

    name "Test Process STD_PRIMER_FLAG"
    script "modules/local/std_primer_flag/main.nf"
    process "STD_PRIMER_FLAG"
    tag "modules"
    tag "std_primer_flag"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    "16S-V3-V4",
                    file('${projectDir}/modules/local/std_primer_flag/tests/data/test.fastq.gz', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.std_primer_out).match("std_primer_out")
        }

    }

}
