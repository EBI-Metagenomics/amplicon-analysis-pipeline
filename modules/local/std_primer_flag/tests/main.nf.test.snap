{
    "std_primer_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "16S-V3-V4",
                    "test_16S-V3-V4_std_primers.fasta:md5,f26cdc22a5231d2dbd0831f1f1983dc6"
                ]
            ]
        ],
        "timestamp": "2023-11-03T11:38:07.362886238"
    }
}